### DISCUSSION TITLE
Auto Connect with Bluetooth?

### DISCUSSION THREAD

**Discussion by ransa808** [tttapa/Control-Surface#393] [General]
Hi

Every time I power on my device I need to go the OSX Midi Bluetooth config and connect with the device manually.
Is it possible to force the device to connect to my mac once it's powered on?

Thx

**Comment by tttapa** [comment]
My Ubuntu system connects automatically each time, I don't use OSX myself, so I'm afraid I can't really help you with that. 

The master version of Control Surface uses the [ESP32 BLE library](https://github.com/espressif/arduino-esp32/tree/master/libraries/BLE), while the `new-input` branch uses the low-level [Espressif BLE API](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/bluetooth/bt_le.html). To understand it, you'll probably have to look at the ESP-IDF examples first. You can find links to the relevant ones in [this README](https://github.com/tttapa/Control-Surface/tree/new-input/src/MIDI_Interfaces/BLEMIDI/ESP32). The code for the connection events is in https://github.com/tttapa/Control-Surface/blob/new-input/src/MIDI_Interfaces/BLEMIDI/ESP32/events.c.  
You should probably also read chapter 36 of the [Apple Accessory Design Guidelines](https://developer.apple.com/accessories/Accessory-Design-Guidelines.pdf). I don't own any Apple devices myself, so I can't test it.

**Reply by ransa808** [reply]
Thanks a lot for everything buddy, I'll have a look

**Reply by ransa808** [reply]
This is an OSX limitation as it seems... if anyone's interested you can use this Korg app as a shortcut instead of opening the Audio Midi Setup every time: https://support.korguser.net/hc/en-us/articles/207779053-Do-I-have-to-connect-the-microKEY-Air-to-a-Mac-every-time-when-it-is-turned-on-

BTW - Is there an option to change the BLE Device Name ?

**Reply by tttapa** [reply]
Master:
https://github.com/tttapa/Control-Surface/blob/feee931d44d5f20fb7913b171895316e5819549d/src/MIDI_Interfaces/BLEMIDI.hpp#L22

New-input:
https://github.com/tttapa/Control-Surface/blob/33ae4d319b4842d2cfcb3b16767f83c922e97b67/src/MIDI_Interfaces/BLEMIDI/ESP32/midi-init.c#L53

**Comment by jhsa** [comment]
Hi, I am having the same problem on Android. I have to manually connect to the BLE module every time it is disconnected. It doesn't seem to connect automatically..
Thanks.
